tm/src/main/java/io/seata/tm/api/TransactionalTemplate.java

```diff
@@ -91,6 +92,10 @@ private void completeTransactionAfterThrowing(TransactionInfo txInfo, GlobalTran
                 throw new TransactionalExecutor.ExecutionException(tx, txe,
                         TransactionalExecutor.Code.RollbackFailure, ex);
             }
+            //throw to Launcher
+            if (tx.getRole() == GlobalTransactionRole.Participant) {
+                throw new TransactionException(TransactionExceptionCode.ParticipantReportRollback);
```

Comment:
please throw original business exception.

Reply:
> please throw original business exception.

changed

Reply:
original business exception is ex.
if rollbackDone, we need throw ex.